### ISSUE TITLE
Optionally return the digest of the manifest

### ISSUE THREAD

**Issue by shizhMSFT** [deislabs/oras#68]
Optionally return the digest of the manifest on push and pull.

**Comment by shizhMSFT** [comment]
As @sajayantony suggested, we should always display the tag digest on pull and push when using CLI

**Comment by sajayantony** [comment]
ORAS aligns well with content addressability. A user is able to pull by tag or by digest. Enabling the user to know the digest of the content allows the user to pin the remaining commands against that digest instead of a mutable tag.

**Comment by shizhMSFT** [comment]
The current API interface is not suitable for passing the tag digest. Public API changes will apply to `oras.Pull` and `oras.Push`.

/cc @jdolitsky

**Comment by jdolitsky** [comment]
This sounds good. Question somewhat related to this @shizhMSFT 

Is it possible right now with oras to determine digests at the individual layer level prior to fetching the actual content from remote?

The use case being Helm. Broken into 2 layers - chart and meta json - we dont necessarily want to initiate transfer if, for example, the chart content already exists on disk.

You can see here how we are just printing "Chart is up to date" despite having already downloaded the content, regardless of whether or not it already existed on disk: 
https://github.com/helm/helm/blob/dev-v3/pkg/registry/client.go#L93

Here is the source of the `StoreReference` method referenced there if you want to take a look: https://github.com/helm/helm/blob/dev-v3/pkg/registry/cache.go#L172

Is this just a matter of using something other than MemoryStore? If this is not currently supported (and there is a clear path of how to support) I can open a new issue to address.

**Comment by shizhMSFT** [comment]
@jdolitsky It is possible for the caller to determine whether to fetch a child or not before actual fetching it from the remote.

However, the pain point is that the fetching process is done in parallel by `containerd` as for performace considerations. The caller needs to manually sync the sequence, i.e. make sure meta json is fetched before chart. Another approach is that we can have an option that syncly fetching everythin one by one without parallelism.

**Comment by sajayantony** [comment]
How about a fetch filter? For e.g. we could support an annotation based filter where layer matches the annotation.

**Comment by shizhMSFT** [comment]
That's exactly what I think to address the first issue. But for the real scenario, we need to fetch content synchronously.

**Comment by shizhMSFT** [comment]
Further discussion on #75, #76
